jsonata.js

```diff
@@ -2013,6 +2013,137 @@ function functionUppercase(str) {
 }
 
 /**
+ * length of a string
+ * @param {String} str - string
+ * @returns {Number} The number of characters in the string
+ */
+function functionLength(str) {
+    if(arguments.length != 1) {
+        throw {
+            message: 'The length function expects one argument',
+            stack: (new Error()).stack
+        };
+    }
+
+    // undefined inputs always return undefined
+    if(typeof str === 'undefined') {
+        return undefined;
+    }
+
+    // otherwise it must be a string
+    if(typeof str !== 'string') {
+        throw {
+            message: 'Type error: argument of length function must evaluate to a string',
+            stack: (new Error()).stack,
+            value: str
+        };
+    }
+
+    return str.length;
+}
+
+/**
+ * Split a string into an array of substrings
+ * @param {String} str - string
+ * @param {String} separator - the token that splits the string
+ * @param {Integer} limit - max number of substrings (optional)
```

Comment:
You can use `[limit]` to indicate to JSDoc that the `limit` parameter is optional